applications/ShapeOptimizationApplication/custom_utilities/geometry_utilities.h

```diff
@@ -196,6 +206,68 @@ class GeometryUtilities
     	KRATOS_CATCH("");
     }
 
+    // --------------------------------------------------------------------------
+    void ComputeProjectedDistancesToBoundingModelPart(
+        ModelPart& rBoundingModelPart,
+        pybind11::list& rSignedDistances,
+        pybind11::list& rDirections )
+    {
+        KRATOS_TRY;
+
+        // Type definitions for tree-search
+        typedef Node < 3 > NodeType;
+        typedef NodeType::Pointer NodeTypePointer;
+        typedef std::vector<NodeType::Pointer> NodeVector;
+        typedef std::vector<NodeType::Pointer>::iterator NodeIterator;
+        typedef std::vector<double>::iterator DoubleVectorIterator;
+        typedef Bucket< 3, NodeType, NodeVector, NodeTypePointer, NodeIterator, DoubleVectorIterator > BucketType;
+        typedef Tree< KDTreePartition<BucketType> > KDTree;
+
+        size_t bucket_size = 1000;
```

Comment:
Why bucket size 1000?

Reply:
fixed - set to 100 as in our other files.